homeassistant/components/rabbitair/config_flow.py

```diff
@@ -0,0 +1,125 @@
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers import device_registry as dr
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def validate_input(hass: HomeAssistant, data: dict[str, Any]) -> dict[str, Any]:
+    """Validate the user input allows us to connect."""
+    try:
+        try:
+            zeroconf_instance = await zeroconf.async_get_async_instance(hass)
+            with UdpClient(
+                data[CONF_HOST], data[CONF_ACCESS_TOKEN], zeroconf=zeroconf_instance
+            ) as client:
+                info = await client.get_info()
+        except Exception as err:
+            _LOGGER.debug("Connection attempt failed: %s", err)
+            raise
+    except ValueError as err:
+        # Most likely caused by the invalid access token.
+        raise InvalidAccessToken from err
+    except OSError as err:
+        # Most likely caused by the invalid host.
+        raise InvalidHost from err
+    except asyncio.TimeoutError as err:
+        # Either the host doesn't respond or the auth failed.
+        raise TimeoutConnect from err
+    except Exception as err:
+        # Other possible errors.
+        raise CannotConnect from err
+
+    # Return info to store in the config entry.
+    return {"title": f"RabbitAir-{info.mac.replace(':', '')}", "mac": info.mac}
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Rabbit Air."""
+
+    VERSION = 1
+
+    _discovered_host: str | None = None
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        errors = {}
+
+        if user_input is not None:
+            try:
+                info = await validate_input(self.hass, user_input)
+            except CannotConnect:
+                errors["base"] = "cannot_connect"
+            except InvalidAccessToken:
+                errors["base"] = "invalid_access_token"
+            except InvalidHost:
+                errors["base"] = "invalid_host"
+            except TimeoutConnect:
+                errors["base"] = "timeout_connect"
+            except Exception as err:  # pylint: disable=broad-except
+                _LOGGER.debug("Unexpected exception: %s", err)
+                errors["base"] = "unknown"
+            else:
+                await self.async_set_unique_id(dr.format_mac(info["mac"]))
+                self._abort_if_unique_id_configured(updates=user_input)
+                return self.async_create_entry(title=info["title"], data=user_input)
+
+        user_input = user_input or {}
+        host = user_input.get(CONF_HOST, self._discovered_host)
+        token = user_input.get(CONF_ACCESS_TOKEN)
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_HOST, default=host): str,
```

Comment:
To make you life easier (in terms of typing) down the line, you might want to consider putting the MAC address in the config entry data as well.
Right now, you seem to rely the config entry unique ID in those cases, that works, but since unique ID is optional typed, that makes it a tiny bit harder.

Reply:
Implemented. The MAC address is now stored in the config entry data.